shelley/chain-and-ledger/formal-spec/chain.tex

```diff
@@ -746,15 +789,42 @@ \subsection{Block Header Transition}
 \subsection{Operational Certificate Transition}
 \label{sec:oper-cert-trans}
 
-The Operational Certificate Transition contains no environments, and its state is the mapping of
-operation certificate issue numbers.  Its signal is a block header.
+The Operational Certificate Transition environment consists of the genesis key
+delegation map $\var{genDelegs}$ and the set of stake pools $\var{stpools}$. Its state
+is the mapping of operation certificate issue numbers.  Its signal is a block
+header.
 
 \begin{figure}
+  \emph{Operational Certificate environments}
+  \begin{equation*}
+    \OCertEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{stpools} & \powerset{\type{KeyHash}} & \text{stake pools} \\
+        \var{genDelegs} & \powerset{\type{KeyHash}} & \text{genesis key delegates}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
   \emph{Operational Certificate Transitions}
   \begin{equation*}
-    \vdash \var{\_} \trans{ocert}{\_} \var{\_} \subseteq
-    \powerset (\KeyHash_{pool} \mapsto \N \times \BHeader \times \KeyHash_{pool} \mapsto \N)
+    \var{\_} \vdash \var{\_} \trans{ocert}{\_} \var{\_} \subseteq
+    \powerset (\OCertEnv \times \KeyHash_{pool} \mapsto \N \times \BHeader \times \KeyHash_{pool} \mapsto \N)
   \end{equation*}
+  %
+  \emph{Operational Certificate helper function}
+  \begin{align*}
+      & \fun{currentIssueNo} \in \OCertEnv \to (\KeyHash_{pool} \to \N)
+                                           \to \KeyHash_{pool}
+                                           \to \N^? \\
+      & \fun{currentIssueNo}~(\var{stpools}, \var{genDelegs})~ \var{cs} ~\var{hk} \\
+      & \begin{array}{ccl}
+        ~~~~ & | & \var{hk}\mapsto \var{n} \in \var{cs} = n \\
```

Comment:
instead of using haskell notation here, you could use `cases`, something like:
```
      & \fun{currentIssueNo}~(\var{stpools}, \var{genDelegs})~ \var{cs} ~\var{hk} = 
        \begin{cases}
          \var{hk}\mapsto \var{n} \in \var{cs} & n \\
          \var{hk} \in \var{stpools} & 0 \\
          \var{hk} \in \var{genDelegs} & 0 \\
          \text{otherwise} & \Nothing
        \end{cases}
```